packages/grafana-e2e/src/support/scenarioContext.ts

```diff
@@ -1,29 +1,36 @@
+import { e2e } from '../index';
+
 export interface ScenarioContext {
-  lastAddedDataSource: string;
   lastAddedDashboard: string;
   lastAddedDashboardUid: string;
+  lastAddedDataSource: string;
+  lastAddedDataSourceId: string;
   [key: string]: any;
 }
 
-const scenarioContexts: ScenarioContext = {
-  lastAddedDataSource: '',
+const scenarioContext: ScenarioContext = {
   lastAddedDashboard: '',
   lastAddedDashboardUid: '',
+  lastAddedDataSource: '',
+  lastAddedDataSourceId: '',
 };
 
-export interface ScenarioContextApi {
-  get: <T>(name: string | keyof ScenarioContext) => T;
-  set: <T>(name: string | keyof ScenarioContext, value: T) => void;
-}
-
-export const scenarioContext = (): ScenarioContextApi => {
-  const get = <T>(name: string | keyof ScenarioContext): T => scenarioContexts[name] as T;
-  const set = <T>(name: string | keyof ScenarioContext, value: T): void => {
-    scenarioContexts[name] = value;
-  };
+// @todo this actually returns type `Cypress.Chainable`
+export const getScenarioContext = (): any =>
+  e2e()
+    .wrap({
+      getScenarioContext: () => ({ ...scenarioContext } as ScenarioContext),
+    })
+    .invoke('getScenarioContext');
```

Comment:
There were timing issues because the context function was not part of Cypress' resulting asynchronous queue. Imagine this:
```js
log(getContext('key')); //-> 'A'
somethingAsynchronous().then(() => setContext('key', 'B'));
log(getContext('key')); //-> 'A'
```

Reply:
Gotcha.